src/test/userRepository.test.ts

```
@@ -0,0 +1,166 @@
+}
+
+describe("create user", () => {
+  beforeAll(async () => {
+    await connectDatabase();
+  });
+
+  beforeEach(async () => {
+    await UserModel.deleteMany({});
+  });
+
+  afterAll(async () => {
+    await UserModel.deleteMany({});
+    await closeConnectionDatabase();
+  });
+
+  test("should fail: undefined email", async () => {
+    const [, error] = await userRepository.create({ password: MOCK_PASSWORD });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_UNDEFINED);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: empty email", async () => {
+    const [, error] = await userRepository.create({
+      email: "",
+      password: MOCK_PASSWORD,
+    });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_EMPTY);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: invalid email", async () => {
+    const [, error] = await userRepository.create({
+      email: "abc",
+      password: MOCK_PASSWORD,
+    });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_INVALID);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: undefined password", async () => {
+    const [, error] = await userRepository.create({ email: MOCK_EMAIL });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.PASSWORD_UNDEFINED);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.PASSWORD]);
+  });
+
+  test("should fail: password too short", async () => {
```

Maybe "should throw an error"